x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -410,7 +414,51 @@ func (m *MongoCrypt) createErrorFromStatus() error {
 	return errorFromStatus(status)
 }
 
+// needsKmsProvider returns true if provider was initially set to an empty document.
+// An empty document signals the driver to fetch credentials.
+func needsKmsProvider(kmsProviders bsoncore.Document, provider string) bool {
+	val, err := kmsProviders.LookupErr(provider)
+	if err != nil {
+		// KMS provider is not configured.
+		return false
+	}
+	doc, ok := val.DocumentOK()
+	// KMS provider is an empty document.
+	return ok && len(doc) == 5
```

Comment:
What is the significance of a document length of 5?

Reply:
An empty document contains 4 bytes of "\x00" and a null byte.